internal/cloud/registry/ecr/ecr.go

```diff
@@ -0,0 +1,62 @@
+package ecr
+
+import (
+	"context"
+	"regexp"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/aws/aws-sdk-go-v2/service/ecr/ecriface"
+	"github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+
+	"github.com/dominodatalab/forge/internal/cloud/registry"
+)
+
+var (
+	urlRegex = regexp.MustCompile(`^\d{12}\.dkr\.ecr\.[a-z0-9-]+\.amazonaws.com$`)
```

Comment:
https://github.com/awslabs/amazon-ecr-credential-helper/blob/10153a961e2dd9da9f40f6cc284572252b199c58/ecr-login/api/client.go#L34 is a bit more complicated. I wonder if in general the best way to find the "correct" loader is to just try and grab a token.

Reply:
It seems like using the regex pattern from the ECR helper is a better option than the one I wrote. That should be an improvement and easy to write tests against that. 

>I wonder if in general the best way to find the "correct" loader is to just try and grab a token.

I guess I don't understand what you mean when you say "just try". Are you suggesting we don't use regex but instead try and perform auth using all loaders until we hit gold? If so, considering the following: (a) blind auth calls can be expensive (e.g. AWS config loading will hang for several secs if there are no valid credentials, not sure about the others) and (b) I'm not sure what the errors returned by the other libraries are (i.e. how do we discern between an error that indicates "not supported" and "supported but failed to auth"?).

Reply:
Yeah that was the gist. I hadn't considered the performance cost and the potential error handling constraints, thanks.